### DISCUSSION TITLE
tutorials/lift-off-part1/03-schema-definition-language-sdl

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#38] [course feedback]
# tutorials/lift-off-part1/03-schema-definition-language-sdl

https://www.apollographql.com/tutorials/lift-off-part1/03-schema-definition-language-sdl

**Comment by CMH7** [comment]
Super easy to follow loving it <3

**Comment by Theedon** [comment]
I am liking the way the tutorial is structured so far

**Comment by yunushabeeb** [comment]
Short, concise, and practicable. I am already crushing on the tutorial.

**Comment by dilipagheda** [comment]
to the point. no unnecessary stories. really good!

**Comment by zichengshan** [comment]
So far so Good!

**Comment by 4Furki4** [comment]
one of the best guides I've ever seen

**Comment by jdanna204** [comment]
Excellent course; kudos to the content creators and the back-end developers who enabled the scripts

**Comment by brencon** [comment]
Love being able to write code within the lesson and have that code validated. Awesome!

**Comment by neogeek13** [comment]
Had a really strange experience. I was not logged in to my GH account and answered the question
"Which of the following are valid field types?" thinking it was asking for ALL potential correct types. I got it wrong with the 'correct' answer being only Int and String! (implying the question should have been "Which of the following are valid field types for this Graph?").

When I came to post suggesting the question text be changed, it cleared my response. On retaking the quiz now that I'm logged in, NOW it wants ALL boxes checked as it's my original interpretation - ALL of the options are all valid field types in general. I can't imagine there are two versions of that question based on login but just in case... (P.S. Yes, I'm a Test Engineer... :P )

**Reply by mabuyo** [reply]
Hey @neogeek13 ! Thanks for the detailed information, that's very helpful! Unfortunately I can't reproduce what you encountered. The correct answers _should_ be **all of them** and we haven't changed that question recently. It does make sense that your response was cleared, because we don't save responses when the user isn't logged in. 

We'll keep an eye out if this happens again, but can confirm that the question text and the answers are correct :)

**Reply by neogeek13** [reply]
Thanks, I figured that's how it would go. Was super strange and if I hadn't checked three times I would have been sure I just imagined it. XD

**Comment by saiyantanghosh** [comment]
done

**Comment by doanwebute** [comment]
very nice to start graphql

**Comment by teerit** [comment]
very good lesson and example

**Comment by EjDadivas** [comment]
The exercises are really good to enforce what we learned! Nice interactive course

**Reply by EjDadivas** [reply]
This is a good way to introduce someone to GraphQL

**Comment by EjDadivas** [comment]
What I learned:
- type *TypeName* - using type keyword to define a type
- inside the type body you can define fields
-fields could be scalar or objects (other types)
- ! - not nullable,  [*FieldType*] - array of that certain field type
- Adding descriptions: """block """, "normal"

**Reply by mabuyo** [reply]
Good summary! Glad you're enjoying the course :)

**Comment by onesamket** [comment]
it's amazing  I love it ❤️❤️

**Comment by Tejeswar437** [comment]
Its easy understand . loved it

**Comment by repiemm** [comment]
Loved the code challenges 👌🏽

**Comment by belloshehu** [comment]
I always enjoy the moment of defining my data, query, mutation and subscription using SDL. It is simple but a very powerful tool!

**Comment by bipin1981** [comment]
Love it

**Comment by LingVuDev** [comment]
Really enjoy the tutorial so far.

**Comment by NishantCoder108** [comment]
# Schema Definition Language (SDL)
_________________________________________

## Important points i learn:
- ID , String , Int, Boolean , [Int] , Float, these are types , that are mostly used.
- we make schema like
   ```type PascalCase {
      age : Int
      name: String!
     mission: [String]
}

**Comment by ikoral** [comment]
Very cool, practical

**Comment by vishwaefor** [comment]
Shouldn't the missions be expecting [Mission!]
```
type SpaceCat {
  name: String!
  age: Int
  missions: [Mission!]
}
```

**Reply by mabuyo** [reply]
Hi @vishwaefor, are you referring to the code challenge? The instructions specifically say:

> `missions` of type `List` of `Mission`

**Reply by vishwaefor** [reply]
Thanks for the reply. It means the List of Missions can have null as a value.

**Comment by dtoxvanilla1991** [comment]
I agree with others, I am loving the structure.

**Comment by khalidadamu** [comment]
Loving it, very much practical

**Comment by mirzasahin** [comment]
Clear and concise instructions.. Great resource for learning!

**Comment by gustvoassuncao** [comment]
So clear and so well for us, amazing!

**Comment by RomanyU1662160** [comment]
Very good, thank you

**Comment by huy312100** [comment]
Clear and easy to follow. Great work!

**Comment by acidless** [comment]
I got 500 error on all iframes with code challenges

**Reply by mabuyo** [reply]
Hey there, thanks for reporting the bug! It should be fixed now if you give it another try.

**Reply by acidless** [reply]
Yes, now it works. Thank you!

**Comment by omooladev** [comment]
I have just learned how do document a graphql schema. Thank you

**Reply by omooladev** [reply]
I meant how to***
